### PR TITLE
added delete button

### PR DESCRIPTION
**Pull request by JauniusZ**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -63,21 +63,39 @@ function renderTodos() {
 
   filteredTodos.forEach((todo, index) => {
     const listItem = document.createElement("li");
+    listItem.style.display = "flex";
```
Comment (arturasmckwcz) [inline script.js]: this is inline styling actually which complicates specificity and is a bad practice IMHO. next time you open your css file you won't find that there, but it still has an effect on styling.

File: script.js [hunk 2]
```diff
@@ -63,21 +63,39 @@ function renderTodos() {
...
+
     const checkbox = document.createElement("input");
     checkbox.type = "checkbox";
     checkbox.checked = todo.checked;
     listItem.appendChild(checkbox);
 
     const titleText = document.createElement("span");
     titleText.textContent = todo.title;
+    titleText.style.flex = "1";
```
Comment (arturasmckwcz) [inline script.js]: inline styling!

File: script.js [hunk 3]
```diff
@@ -63,21 +63,39 @@ function renderTodos() {
...
     listItem.appendChild(titleText);
 
+    const deleteButton = document.createElement("img");
+    deleteButton.src =
+      "https://pluspng.com/img-png/delete-button-png-delete-icon-1600.png";
+    deleteButton.alt = "Delete";
+    deleteButton.style.width = "20px";
+    deleteButton.style.height = "20px";
```
Comment (arturasmckwcz) [inline script.js]: inline styling!

File: script.js [hunk 4]
```diff
@@ -63,21 +63,39 @@ function renderTodos() {
...
+    listItem.appendChild(deleteButton);
+
     checkbox.addEventListener("change", () => {
       state.todos[index].checked = checkbox.checked;
       updateUncheckedTodoCount();
       sessionStorage.setItem(SESSIONSTORAGE_KEY, JSON.stringify(state.todos));
     });
 
+    deleteButton.addEventListener("click", () => {
+      state.todos.splice(index, 1);
```
Comment (arturasmckwcz) [inline script.js]: this is a bit complicated 🤫
despite it works perfectly you'd better keep your state immutable, so you need to produce a new updated array and assign it to the state:
```
state.todos = [...state.todos].splice(index, 1);
```

**Review by arturasmckwcz — COMMENTED** [review]
please move inline styling to css file or just remove it from the script and leave that for the next ticket.

regarding immutability of the state. in such a simple app as todos it doesn't make any difference, but if you'd follow the concept of a stateful app you need to understand why you might have to take such an effort

File: script.js (hunk 1 above)
Comment (JauniusZ) [reply script.js]: Moved inline styling to css file.

File: script.js (hunk 2 above)
Comment (JauniusZ) [reply script.js]: Likewise.

File: script.js (hunk 3 above)
Comment (JauniusZ) [reply script.js]: Likewise.

File: script.js (hunk 4 above)
Comment (JauniusZ) [reply script.js]: Could you explain immutability of the state in a bit more detail?

Comment (arturasmckwcz) [reply script.js]: Okay you have already noticed processes in a browser are of asynchronous nature haven't you? And global state might be used by several processes simultaneously. If it is mutable then some process might experience discrepancies caused by changes of a state during its run. Imagine a large scale app which has an array of something as a state, okay? And on of its processes is doing heavy calculations on the state. And during that time consuming process a user clicks a button which deletes an element of the state array. If it's not immutable those calculations may fail and error out. If it's immutable then clicking of delete button would crest a new state array while the process of heavy calculations would be still dealing with the previous version of the state array. Not sure if I'm clear enough here, please ask me on an occasion for more details okay?

Comment (JauniusZ) [reply script.js]: ChatGPT suggests creating two subarrays:
 
` state.todos = [...state.todos.slice(0, index), ...state.todos.slice(index + 1)];`

What do you think?

Comment (arturasmckwcz) [reply script.js]: It's the same: either you create a copy of an array and remove an element from it or you create two arrays excluding the element you needed to remove and then concatenate them. The first looks a bit clearer to me.

The point here is the difference between `slice` and `splice`. The first returns a new array while the second applies changes to the provided one, i.e. mutates it. As objects, an array is an object in JavaScript, are passed as args by reference not by value. This means that an object can possibly be mutated by one process while it is being used by another.

Comment (JauniusZ) [reply script.js]: `state.todos = [...state.todos].splice(index, 1);` 

This seems to have the opposite effect and delete all the todos except the one that is clicked.

Comment (arturasmckwcz) [reply script.js]: 😀😀😀
sure `splice` returns deleted items. Should be:
```
const newTodos = [...state.todos];
newTodos.splice(index, 1);
state.todos = newTodos;
```

**Pull request merged**